AppLovinAdapter/src/main/java/com/chartboost/mediation/applovinadapter/AppLovinAdapter.kt

```diff
@@ -52,79 +52,115 @@ class AppLovinAdapter : PartnerAdapter {
 
                     adInfo?.let { adId ->
                         withContext(Main) {
-                            AppLovinSdk.getInstance(context).settings.testDeviceAdvertisingIds =
-                                listOf(adId)
+                            appLovinSdk?.let { sdk ->
+                                sdk.settings.testDeviceAdvertisingIds = listOf(adId)
+
+                                PartnerLogController.log(
+                                    CUSTOM,
+                                    "AppLovin test mode is enabled. Remember to disable it before publishing."
+                                )
+                            } ?: run {
+                                PartnerLogController.log(
+                                    CUSTOM,
+                                    "Unable to set test mode. AppLovin SDK instance is null."
+                                )
+                            }
                         }
                     } ?: run {
-                        PartnerLogController.log(
-                            CUSTOM,
-                            "AppLovin test mode is disabled. No advertising id found."
-                        )
-                        AppLovinSdk.getInstance(context).settings.testDeviceAdvertisingIds =
-                            emptyList()
+                        appLovinSdk?.let { sdk ->
```

Comment:
you are doing the first section withContext(Main) but this code is similar but doesn't have it. You're not the one touching that behavior but just noticing it.